### PR TITLE
make a custom io that throws when too much been written to it

### LINKED ISSUE #236: Debugger hangs on very large arrays
Due to the call to `repr` 
```julia
julia> @enter length(view(1:100_000_000,:)) # hangs for a long time
```
This was previously discussed in https://github.com/JuliaDebug/Debugger.jl/issues/65 but I now realised it is triggered by Base types as above.

### PR DESCRIPTION
**Pull request by KristofferC**
this helps with delay when the debugger tries to show many large values

Fixes #236

### CODE REVIEW

File: src/printing.jl [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+    iob = IOBuffer()
+    local limited_str
+    try
+        limit_io = LimitIO(iob, n)
+        f(limit_io, val)
+        limited_str = String(take!(iob))
+    catch e
+        if e isa LimitIOException
+            limited_str = String(take!(iob))
+            limited_str = suppressed("$limited_str")
```
Comment (pfitzseb) [inline src/printing.jl]:
```suggestion
            limited_str = suppressed(limited_str)
```

Comment (KristofferC) [reply src/printing.jl]: Hah, thanks!

**Comment by timholy** [comment]
Very elegant solution, thanks for doing this!

**Comment by KristofferC** [comment]
Maybe this is a bit of an "abuse" of exceptions. I wonder if overloading `eof` would work.

**Pull request merged**